tgui/packages/tgui/interfaces/ExperimentConfigure.js

```diff
@@ -0,0 +1,177 @@
+
+export const ExperimentStage = props => {
+  const [type, description, value, altValue] = props;
+
+  // Determine completion based on type of stage
+  let completion = false;
+  switch (type) {
+    case "bool":
+      completion = value;
+      break;
+    case "integer":
+      completion = value === altValue;
+      break;
+    case "float":
+      completion = value >= 1;
+      break;
+  }
+
+  return (
+    <div className={`ExperimentStage__StageContainer ${completion ? "complete" : "incomplete"}`}>
+      <Flex>
+        <Flex.Item
+          className={`ExperimentStage__Indicator ${type}`}
+          color={completion ? "good" : "bad"}>
+          {(type === "bool" && <Icon name={value ? "check" : "times"} />)
+            || (type === "integer" && `${value}/${altValue}`)
+            || (type === "float" && `${value * 100}%`)}
+        </Flex.Item>
+        <Flex.Item className="ExperimentStage__Description">
+          {description}
+        </Flex.Item>
+      </Flex>
+    </div>
+  );
+};
+
+export const Techweb = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { servers } = props;
+
+  return (
+    <Box m={1} className="ExperimentTechweb__Web">
+      <Flex align="center" justify="space-between"
+        className="ExperimentTechweb__WebHeader">
+        <Flex.Item className="ExperimentTechweb__WebName">
+          {servers[0].web_id} / {servers[0].web_org}
+        </Flex.Item>
+        <Flex.Item>
+          <Button
+            onClick={() => servers[0].selected
+              ? act("clear_server")
+              : act("select_server", { "ref": servers[0].ref })}
+            content={servers[0].selected ? "Disconnect" : "Connect"}
+            backgroundColor={servers[0].selected ? "good" : "rgba(0, 0, 0, 0.4)"}
+            className="ExperimentTechweb__ConnectButton" />
+        </Flex.Item>
+      </Flex>
+      <Box className="ExperimentTechweb__WebContent">
+        <span>
+          Connectivity to this web is maintained by the following servers...
+        </span>
+        <LabeledList>
+          {servers.map((server, index) => {
+            return (
+              <LabeledList.Item
+                key={index}
+                label={server.name}>
+                <i>Located in {server.location}</i>
+              </LabeledList.Item>
+            );
+          })}
+        </LabeledList>
+      </Box>
+    </Box>
+  );
+};
+
+export const ExperimentConfigure = (props, context) => {
+  const { act, data } = useBackend(context);
+  let servers = data.servers ?? [];
+  let experiments = data.experiments ?? [];
+
+  // Toss the selected experiment to the top if it isn't already, sort lexically
+  experiments = experiments.sort((a, b) => {
+    if (a.selected !== b.selected) {
+      return a.selected ? -1 : 1;
+    }
+    else {
+      if (a.name === b.name) {
+        return 0;
+      }
+      else {
+        return a.name < b.name ? -1 : 1;
+      }
+    }
+  });
```

Comment:
For this you need to `import { sortBy } from 'common/collections'`

```suggestion
  const experiments = sortBy(
    exp => exp.selected,
    exp => exp.name
  )(data.experiments ?? []);
```

Reply:
Bumping this.